### PR TITLE
Feature: vote power contract

### PR DESCRIPTION
**Pull request by shark0der**
Exposes an ERC20 interface and returns the amount of tokens held by the owner + balances of all members who delegated. Adds 1 NXM to everyone's balance as per spec. Can only be used against a snapshot as it doesn't check lock times hence doesn't prevent double voting if used on-chain.

### CODE REVIEW

**Comment by danoctavian** [comment]
What's the context for this? where is it needed?

File: contracts/modules/governance/VotePower.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+pragma solidity ^0.8.0;
+
+import "../../interfaces/IGovernance.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/INXMMaster.sol";
+import "../../interfaces/INXMToken.sol";
+import "../../interfaces/ITokenController.sol";
+
+contract VotePower {
```
Comment (danoctavian) [inline contracts/modules/governance/VotePower.sol:11]: have you validated balanceOf and totalSupply is enough for snapshot work ?

I assume it is

File: contracts/modules/governance/VotePower.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    uint delegationId = governance.followerDelegation(member);
+
+    if (delegationId != 0) {
+      (, address leader,) = governance.allDelegation(delegationId);
+      if (leader != address(0)) {
+        return 0;
+      }
+    }
+
+    uint balance = tokenController.totalBalanceOf(member) + 1e18;
```
Comment (danoctavian) [inline contracts/modules/governance/VotePower.sol:44]: what is the reason for the `+ 1e18` ?

File: contracts/modules/governance/VotePower.sol (hunk 1 above)
Comment (shark0der) [reply contracts/modules/governance/VotePower.sol:11]: `balanceOf` is the only function that's being called by the `erc20-balance-of` strategy:
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/erc20-balance-of/index.ts#L8-L10

File: contracts/modules/governance/VotePower.sol (hunk 2 above)
Comment (shark0der) [reply contracts/modules/governance/VotePower.sol:44]: Required by the formula: each wallet votes with amount of nxm held + 1 nxm. In other words it's not a tech-related requirement but a governance rules one.

File: contracts/modules/governance/VotePower.sol [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    uint[] memory delegationIds = governance.getFollowers(member);
+
+    for (uint i = 0; i < delegationIds.length; i++) {
+
+      (address follower, address leader,) = governance.allDelegation(delegationIds[i]);
+
+      if (
```
Comment (danoctavian) [inline contracts/modules/governance/VotePower.sol:51]: there's an extra check in the `_setVoteTally` code:

https://github.com/NexusMutual/smart-contracts/blob/021f721e3e793fa527badf8dd0a297ab5ca74cd2/contracts/modules/governance/Governance.sol#L930

```(now - _lastUpd) > tokenHoldingTime;```

Comment (roxdanila) [reply contracts/modules/governance/VotePower.sol:51]: Snapshot deals with double voting by default, isn't it? So this isn't needed

**Review by roxdanila — APPROVED** [review]

File: contracts/modules/governance/VotePower.sol (hunk 3 above)
Comment (danoctavian) [reply contracts/modules/governance/VotePower.sol:51]: lgtm

**Review by danoctavian — APPROVED** [review]
LGTM

**Pull request merged**